### PR TITLE
 Skip bad log docs.

### PR DESCRIPTION
**Pull request by howbazaar**
## Description of change

The debug-log code assumes that the documents it gets all have well formed tags.
For some strange, yet to be determined, reason, sometimes there is an empty tag.
This branch has debug-log skip those values, and log a warning.

## QA steps

Kinda difficult as the failure seems very intermittent. Did verify that the tests failed first.
```
cd state
go test -check.f LogTailerSuite.TestTailingSkipsBadDocs -check.vv -loggo.location
```
to see the warnings logged for bad tags.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1700453

### CODE REVIEW

**Comment by howbazaar** [comment]
$$merge$$

File: state/logs.go [hunk 1]
```diff
@@ -564,11 +591,20 @@ func (t *logTailer) tailOplog() error {
 			}
 			rec, err := logDocToRecord(t.modelUUID, doc)
 			if err != nil {
-				return errors.Annotate(err, "deserialization failed (possible DB corruption)")
+				if deserialisationFailures == 0 {
+					logger.Warningf("deserialization failed (possible DB corruption), %v", err)
```
Comment (jameinel) [inline state/logs.go]: Should we be surfacing an error to the log channel that we have a bad record? It is a little interesting that we would be writing this to the controller log stream.
		return nil, errors.Annotate(err, "while parsing entity tag")
^ this error could be updated to give any sort of other context of the actual log message, etc. All we get told is "there exists a record that has a missing entity" but it doesn't include the other good fields for the log message for us to know what log record is bad.

Comment (howbazaar) [inline state/logs.go]: Perhaps.

File: state/logs.go [hunk 2]
```diff
@@ -564,11 +591,20 @@ func (t *logTailer) tailOplog() error {
...
+				}
+				deserialisationFailures++
+				continue
+			} else {
+				if deserialisationFailures > 1 {
```
Comment (mjs) [inline state/logs.go]: Doesn't this mean that after the first error it'll keep logging after every received log entry?

Comment (howbazaar) [inline state/logs.go]: It means that it will log every group of failures.

**Pull request merged**